geekbench-calculate-scores.py

```diff
@@ -0,0 +1,148 @@
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
+import json
+import sys
+import numpy
+import scipy
+from scipy import stats

baseline_rate = {"AES" : 2297304827.712579, "Twofish" : 147104771.2752039,
+"SHA1" : 284535727.5317147, "SHA2" : 113421232.21088956,
+"BZip2 Compress" : 10655380.829907415,
+"BZip2 Decompress" : 14207477.374253012,
+"JPEG Compress" :  34828193.30901489,
+"JPEG Decompress" : 61799226.229028925, "PNG Compress" : 1995970.1507709173,
+"PNG Decompress" : 28823191.43475692,
+"Sobel" : 90973382.70326369, "Lua" : 2355476.4774805442,
+"Dijkstra" : 8971474.748035867, "BlackScholes" : 11124707.579848604,
+"Mandelbrot" : 2561771045.08415, "Sharpen Filter" :  1852822632.527193,
+"Blur Filter" : 2382099498.994065,  "SGEMM" : 7001462936.999326,
+"DGEMM" : 3673032244.756834 , "SFFT" : 2635951419.4330635,
+"DFFT" : 2276728468.11824, "N-Body" : 927967.5877615509,
+"Ray Trace" : 2947755.511608884, "Stream Copy" : 10709056870.933207,
+"Stream Scale" : 10717018032.487293, "Stream Add" :  12138602680.677034,
+"Stream Triad" : 11796840626.258623 }
+
+class Workload(object):
+	def __init__(self, jsonf):
		self.name = jsonf['name']
+		if len(jsonf['results']) == 2:
			self.sc_runtime = jsonf['results'][0]['runtime']
			self.mc_runtime = jsonf['results'][1]['runtime']
			self.sc_work = jsonf['results'][0]['work']
			self.mc_work = jsonf['results'][1]['work']
+		elif jsonf['results'][0]['threads'] == 1:
+			self.sc_runtime = jsonf['results'][0]['runtime']
+			self.mc_runtime = 0
+			self.sc_work = jsonf['results'][0]['work']
+			self.mc_work = 0
+		elif jsonf['results'][0]['threads'] > 1:
+			self.sc_runtime = 0
+			self.mc_runtime = jsonf['results'][0]['runtime']
+			self.sc_work = 0
+			self.mc_work = jsonf['results'][0]['work']
+		
	def sc_rate(self):
+		if self.sc_runtime > 0:
			return self.sc_work / self.sc_runtime
		return 0
+		
	def mc_rate(self):
		if self.mc_runtime > 0:
+			return self.mc_work / self.mc_runtime
		return 0
+		
	def wl_sc_score(self):
		rel = self.sc_rate() / baseline_rate[self.name]
		return int(2500 * rel)
	
+	def wl_mc_score(self):
+		rel = self.mc_rate() / baseline_rate[self.name]

```

Comment:
You can get rid of some of the chained indexing operations by doing something like:

self.sc_runtime = 0
self.sc_work = 0
self.mc_runtime = 0
self.mc_work = 0

for result in jsonf['results']:
  if result['threads'] == 1:
    self.sc_runtime = result['runtime']
    self.sc_work = result['work']
  elif result['threads'] > 1:
    self.mc_runtime = result['runtime']
    self.mc_work = result['work']
  else:
    raise ValueError("Result had zero worker threads for {}".format(name))